API/projectClass/layersClass.m

```diff
@@ -191,18 +191,19 @@
             % The expected inputs are the potential layer parameter value
             % (either name or index) and a list of parameter names.
 
-            if isText(inputVal)
-                if ~any(strcmpi(inputVal, paramNames))
+            paramNames = cellstr(paramNames);
+            if isText(inputVal)    
+                found = strcmpi(inputVal, paramNames);
+                if ~any(found)
                     throw(exceptions.nameNotRecognised(sprintf('Parameter %s not recognized', inputVal)));
                 end
-                param = inputVal;
+                param = paramNames{find(found, 1)};
 
             elseif isnumeric(inputVal)
-                paramIndex = floor(inputVal);
```

Comment:
This line presumably ensure the numeric input is an integer. Are you happy that it isn't necessary?

Reply:
Yes, its a bad idea as we cannot assume the users intentions, if `inputVal` is 1.5, does the user want 1 or 2? Also its not done anywhere else.